### ISSUE TITLE
PowerShell 5.1 Failure to Deploy

### ISSUE THREAD

**Issue by mahacket** [OfficeDev/microsoft-teams-apps-company-communicator#728]
Having issues with the most recent version of deployment via PowerShell - attempted deployment 4 times in different regions. Resources continually fail due to BadRequest and Conflict. Screenshot and logs provided. 

![image](https://user-images.githubusercontent.com/95437258/169066967-d5d0caec-6c42-4213-bd44-80456a1cfba3.png)

[deployment_operation.log](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/files/8717175/deployment_operation.log)

**Comment by rramos9771** [comment]
I have the same problem, app service never synchronizes and does not send a description of the error.

**Comment by satswere** [comment]
at the time of doing the implementation with PowerShell I get that same error

![Captura](https://user-images.githubusercontent.com/67810335/169138676-8d9b8821-60cf-4e7f-aa81-82d99fbb18f5.PNG)


LOG :


Command: deploy.cmd
Handling ASP.NET Core Web Application deployment.
Restoring NuGet packages

Welcome to .NET Core 3.1!
---------------------
SDK Version: 3.1.417

Telemetry
---------
The .NET Core tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of telemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Find out what's new: https://aka.ms/dotnet-whats-new
Learn about the installed HTTPS developer cert: https://aka.ms/aspnet-core-https
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli-docs
Write your first app: https://aka.ms/first-net-core-app
--------------------------------------------------------------------------------------
  Determining projects to restore...
  Restored D:\home\site\repository\Source\Test\CompanyCommunicator.Test\Microsoft.Teams.Apps.CompanyCommunicator.Test.csproj (in 1.68 min).
  Restored D:\home\site\repository\Source\CompanyCommunicator.Send.Func\Microsoft.Teams.Apps.CompanyCommunicator.Send.Func.csproj (in 1.68 min).
  Restored D:\home\site\repository\Source\CompanyCommunicator.Prep.Func\Microsoft.Teams.Apps.CompanyCommunicator.Prep.Func.csproj (in 5.36 sec).
  Restored D:\home\site\repository\Source\CompanyCommunicator.Data.Func\Microsoft.Teams.Apps.CompanyCommunicator.Data.Func.csproj (in 5.29 sec).
  Restored D:\home\site\repository\Source\Test\CompanyCommunicator.Send.Func.Test\Microsoft.Teams.Apps.CompanyCommunicator.Send.Func.Test.csproj (in 2.19 sec).
  Restored D:\home\site\repository\Source\Test\CompanyCommunicator.Prep.Func.Test\Microsoft.Teams.Apps.CompanyCommunicator.Prep.Func.Test.csproj (in 1.67 sec).
  Restored D:\home\site\repository\Source\Test\CompanyCommunicator.Common.Test\Microsoft.Teams.App.CompanyCommunicator.Common.Test.csproj (in 1.61 sec).
  Restored D:\home\site\repository\Source\CompanyCommunicator.Common\Microsoft.Teams.Apps.CompanyCommunicator.Common.csproj (in 9.88 sec).
  Restored D:\home\site\repository\Source\CompanyCommunicator\Microsoft.Teams.Apps.CompanyCommunicator.csproj (in 2.77 sec).
Restoring npm packages (this can take several minutes)

> core-js@2.6.12 postinstall D:\home\site\repository\Source\CompanyCommunicator\ClientApp\node_modules\babel-runtime\node_modules\core-js
> node -e "try{require('./postinstall')}catch(e){}"
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules\webpack-dev-server\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"ia32"})
Command 'starter.cmd deploy.cmd' was aborted due to no output nor CPU activity for 60 seconds. You can increase the SCM_COMMAND_IDLE_TIMEOUT app setting (or WEBJOBS_IDLE_TIMEOUT if this is a WebJob) if needed.\r\nstarter.cmd deploy.cmd
> node -e "try{require('./postinstall')}catch(e){}"
Input string was not in a correct format

**Comment by enito-ch** [comment]
Hey there!

I am having a pretty similar problem. I am also trying to deploy the app in a tenant (counting 6 deployment-attempts and all failed). 

![PrintScreen](https://user-images.githubusercontent.com/103029137/169159352-47a877d8-5818-45f3-8c3a-43ef4e8d4669.png)



Here is the log: [deployment_operation.log](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/files/8721007/deployment_operation.log)

**Comment by gsv022** [comment]
Hi @mahacket ,
We are testing at our end and will get back to you soon.

**Comment by Donavanlr** [comment]
Good day 
Are there any feedback on the deployment issues being experienced. 
Thank you

**Comment by Donavanlr** [comment]
> Hi @mahacket , We are testing at our end and will get back to you soon.

Any feedback on the issues currently being experienced with deployments failing? 

Thank you

**Comment by gsv022** [comment]
@mahacket, @rramos9771 , @enito-ch , @satswere 

The fsevents library is causing the app service deployment to fail. We are working on the fix and will keep you updated. To unblock the deployment please follow the below steps as a quick workaround,

- Go to portal.azure.com. Navigate to resource group where all CC resources are deployed.
- Click on the App service -> Click on Configuration
- Click on **WEBSITE_NODE_DEFAULT_VERSION.**
- Update the default value to **16.13.0** (previous value ->10.15.2)
- Click on Overview and Re-Start the app service.
- Once the app service is restarted. Navigate to **Deployment Center** and click on Sync.
- Wait till the deployment is completed. You can validate this once the status changes to **Success** under logs. 


![image](https://user-images.githubusercontent.com/98143492/170511053-7b2a0cb8-7bc4-477e-b6a6-359012cf6235.png)

![image](https://user-images.githubusercontent.com/98143492/170511111-31010f41-e065-4cd3-a548-075bc61821c3.png)

- For azure function failure, please refer this troubleshooting guide [issue-1] (https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/wiki/Troubleshooting)

**Comment by gsv022** [comment]
Hi @mahacket ,
Could you please confirm whether the above steps have fixed the issue?

**Comment by mahacket** [comment]
Hi @gsv022 and team, 
Yes I was able to successfully get the app deployed with the changes listed above. I did notice with the most recent version now I'm experiencing the same issues as issue 751. (https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/issues/751) 

I believe it has to do with Proactively Installing the user app which I have set to True. I manually installed the app for a single user and did not receive the error for that user and they successfully received the message. For all additional recipients who did not have the user app installed I got the errors listed in issue 751.

**Comment by gsv022** [comment]
Hi @mahacket ,
Some of the customers have reported the similar concern of performance downgrade of Company Communicator and an increase in message failed counts. The behavior is mostly observed for larger recipient count. We have certain limitation in replicating the issue as its dependent on recipient’s size. We are troubleshooting it on priority.



Our initial observation suggests it to be an environment issue, however we are yet to figure out a proper fix. We will keep you updated.

Hoping that the S2 is the app service plan that is used.

**Comment by gsv022** [comment]
Hi @mahacket ,
The connection attempt failed errors was thrown due to the Azure bot services and the issue has been fixed. Could you please re-try sending the messages and share the below observations.

- Volume of target audience
- Time taken to send the messages.
- Connection attempt failed count (If any)

**Comment by enito-ch** [comment]
Hi @gsv022 

Unfortunately, the status remained unchanged (so it's still not working). 
We tried the steps above (https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/issues/728#issuecomment-1138659860) but it did not bring any changes. 

I will try the setup on another tenant and reply to this issue (just to exclude any bad configurations at our end). 

Should you have any new insights or ideas in the meantime please let me know!

**Comment by mahacket** [comment]
Hello @gsv022, 

Yes I was able to send messages successfully now, and did not receive any errors like stated above from issue #751. 

I did a full redeployment yesterday evening and while the source deployment still failed, I was able to get the Company Communicator running by following the steps you provided.

**Comment by gsv022** [comment]
Hi @mahacket ,
Thank you for submitting this issue, as we work to ensure consistent responses with items logged in our issues list, we are working to address these items that did not get responses. Please let us know if you are still facing this issue and require assistance and we will be happy to discuss. If no further action is needed at moment and no response is received, we will close out the issue in 3 business days. You may also always open a new issue or respond here if something does come back up.

**Comment by enito-ch** [comment]
> Hi @gsv022
> 
> Unfortunately, the status remained unchanged (so it's still not working). We tried the steps above ([#728 (comment)](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/issues/728#issuecomment-1138659860)) but it did not bring any changes.
> 
> I will try the setup on another tenant and reply to this issue (just to exclude any bad configurations at our end).
> 
> Should you have any new insights or ideas in the meantime please let me know!

Hey @gsv022 
I was finally able to deploy the app sucessfully! :-) 
What helped: 
- Used the PowerShell ISE console with Win10 (before, I used PowerShell 7.2.3)
- Delete all the resources and wait 24h
- Retried with a complete new resource-group name in the parameters.json file
I am now facing another issue, but I will open a separate issue for that. 
Many thanks for your help guys!
